std/range/package.d

```diff
@@ -1228,6 +1231,382 @@ unittest
     auto c = chain(a, b);
 }
 
+
+/**
+    Choose one of multiple ranges at runtime.
+
+    The ranges may be different, but they must have the same element type. The
+    result is a range that offers the $(D front), $(D popFront), and $(D
+    empty) primitives. If all input ranges offer random access and $(D
+    length), $(D choose) offers them as well.
+
+    Params:
+        index = which range to choose, must be less than the number of ranges
+        rs = tuple of 1 or more ranges
+
+    Returns:
+        The indexed range. If rs consists of only one range,
+        the return type is an alias of that range's type.
+ */
+auto choose(Ranges...)(size_t index, Ranges rs)
```

Comment:
should specialize for Ranges.length == 2 and type of index == bool. That would improve correctness for a common case. BTW in other places (e.g. among or startsWith) we use uint, not size_t, for variadic counts so you may want to continue the tradition.